source/NPC.cpp

```diff
@@ -274,6 +337,22 @@ void NPC::Load(const DataNode &node)
 		ship->SetPersonality(personality);
 		ship->SetIsSpecial();
 		ship->FinishLoading(false);
+		if(!waypoints.empty())
+			ship->SetWaypoints(waypoints);
+		if(!stopovers.empty())
+			ship->SetStopovers(stopovers);
+		if(finalDestination)
+			ship->SetDestination(finalDestination);
+	}
+
+
+	// NPCs given the "land" or "outrun" completion criteria condition should also have a destination.
+	if(!(finalDestination || needsDestination) && destinationFilter.IsEmpty())
+	{
+		if(succeedIf & ShipEvent::LAND)
+			node.PrintTrace("Warning: NPC mission objective to land is impossible without a destination.");
+		else if(failIf & ShipEvent::LAND)
+			node.PrintTrace("Warning: NPC mission objective to outrun is impossible without a destination.");
 	}
```

Comment:
`needsDestination` doesn't seem like a great name in this usage.
This first part of this expression can also be written as:
`!finalDestination && !needsDestination`
So the whole thing becomes:
`!finalDestination && !needsDestination && destinationFilter.IsEmpty()`
Which I would say as:
"There is no final destination, and it does not need a destination, and the destination filter is empty."
So, the whole thing is:
"If there is no final destination, and it does not need a destination, and the destination filter is empty, send a warning that the lack of destination makes the objective impossible."
Which is strange because we're only sending this warning if "it does not need a destination," but the warning is saying it does need a destination?